### PR TITLE
Fix AirLoopHVACUnitarySystem with multispeed coil

### LINKED ISSUE #2278: UnitarySystemPerformance:Multispeed is not cloning correctly
https://unmethours.com/question/18305/unitary-system-performance/

### PR DESCRIPTION
**Pull request by kbenne**
Now OpenStudio will generate UnitarySystemPerformanceMultispeed when
a multispeed heating or cooling coil is used.

close #2278

### CODE REVIEW

File: openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp [hunk 1]
```diff
@@ -455,6 +464,90 @@ boost::optional<IdfObject> ForwardTranslator::translateAirLoopHVACUnitarySystem(
...
+    auto coolingFlow = modelObject.supplyAirFlowRateDuringCoolingOperation();
+
+    for( int i = 0; i < maxStages; ++i ) {
+      auto extensible = _unitarySystemPerformance.pushExtensibleGroup();
+
+      if( i < heatingStages.size() ) {
+        auto heatingStage = heatingStages[i];
+        auto stageFlow = heatingStage.ratedAirFlowRate();
+        if( stageFlow && heatingFlow ) {
+          extensible.setDouble(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio,stageFlow.get() / heatingFlow.get());
```
Comment (mbadams5) [inline openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp]: @kbenne This needs to be a ratio, not flow rate. So probably `heatingFlow.get() / stageFlow.get()` or something like that.

File: openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp [hunk 2]
```diff
@@ -455,6 +464,90 @@ boost::optional<IdfObject> ForwardTranslator::translateAirLoopHVACUnitarySystem(
...
+        }
+      } else if( i < 2 ) {
+        extensible.setDouble(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio,1.0);
+      }
+
+      if( i < coolingStages.size() ) {
+        auto coolingStage = coolingStages[i];
+        auto stageFlow = coolingStage.ratedAirFlowRate();
+        if( stageFlow && coolingFlow ) {
+          extensible.setDouble(UnitarySystemPerformance_MultispeedExtensibleFields::CoolingSpeedSupplyAirFlowRatio,stageFlow.get() / coolingFlow.get());
```
Comment (mbadams5) [inline openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp]: Same as above

File: openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp (hunk 1 above)
Comment (kbenne) [inline openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp]: ummm... isn't that what we have. well the inverse which i think is correct.

Comment (kbenne) [inline openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp]: scroll over. your MacBook Semi-Pro is cutting off some stuff.

**Comment by mbadams5** [comment]
@kbenne When testing added coils to UnitarySystem, I found out that `Coil:Heating:Gas:MultiStage` is considered a multispeed coil in EnergyPlus. This coil should be added to this PR. I have a failing simulation test in openstudio-resources.

**Pull request merged**